web/mrbs_sql.inc

```diff
@@ -97,18 +97,38 @@ function mrbsCheckFree(array $booking, $ignore, $repignore)
   while (false !== ($row = $res->next_row_keyed()))
   {
     unpack_status($row);
-    $starts = getdate($row['start_time']);
+    $start_date = getdate($row['start_time']);
+    $end_date = getdate($row['end_time']);
+    $different_dates = ( $start_date['year'] != $end_date['year'] ||
+                         $start_date['yday'] != $end_date['yday']);
 
     if ($enable_periods)
     {
-      $p_num =$starts['minutes'];
-      $startstr = utf8_strftime($strftime_format['date'] . ", ",
-                                $row['start_time']) . htmlspecialchars($periods[$p_num]);
-    }
-    else
-    {
-      $startstr = utf8_strftime(($twentyfourhour_format) ? $strftime_format['datetime24'] : $strftime_format['datetime12'],
-                                $row['start_time']);
+      $p_num = $start_date['minutes'];
+      $p_end_num = $end_date['minutes'] - 1;
+      if ($different_dates) {
+        $timestr = htmlspecialchars($periods[$p_num]) . ", " .
+                   utf8_strftime($strftime_format['date'], $row['start_time']) . " - " .
+                   htmlspecialchars($periods[$p_end_num]) . ", " .
+                   utf8_strftime($strftime_format['date'], $row['end_time']);
+      } else {
+        $timestr = htmlspecialchars($periods[$p_num]);
+        if ($p_num != $p_end_num) {
+            $timestr .= " - " . htmlspecialchars($periods[$p_end_num]);
+        }
+        $timestr .= ", " . utf8_strftime($strftime_format['date'], $row['start_time']);
+      }
+    } else {
+      if ($different_dates) {
+        $date_format = ($twentyfourhour_format) ? $strftime_format['datetime24'] : $strftime_format['datetime12'];
+      } else {
+        $date_format = ($twentyfourhour_format) ? $strftime_format['time24'] : $strftime_format['time12'];
+      }
+      $timestr = utf8_strftime($date_format, $row['start_time']) . " - " .
+                 utf8_strftime($date_format, $row['end_time']);
+      if (!$different_dates) {
+        $timestr .= ", " . utf8_strftime($strftime_format['date'], $row['start_time']);
```

Comment:
Some of the variable data in $timestr is being escaped, and some isn't - this needs to be fixed.

Reply:
I think the lack of HTML-escaping is due to the fact that it's missing in the main branch, but now would be a good time to fix it anyway.

Reply:
start_date and escaping is now fixed. 

Well, this is embarrassing, I wanted to support you, not introduce problems. :-/ 
I didn't notice the start_date while testing, because I only tested it for the current date. :-(
And yes, the escaping was done as before, only the period names. Now, all timestr are escaped.